model/generate.go

```diff
@@ -392,3 +398,21 @@ func (g *GeneratedProject) validateGeneratedProject(p *Project, cachedProject pr
 
 	return errors.WithStack(catcher.Resolve())
 }
+
+func updateMergeTaskDependencies(p *Project, v *Version) error {
+	execPairs, _, err := p.BuildProjectTVPairsWithAlias(evergreen.CommitQueueAlias)
+	if err != nil {
+		return errors.Wrap(err, "can't get alias pairs")
+	}
+
+	execTaskIDTable := NewTaskIdTable(p, v, "", "").ExecutionTasks
+	mergeTaskDependencies := make([]task.Dependency, 0, len(execPairs))
+	for _, pair := range execPairs {
+		depID := execTaskIDTable.GetId(pair.Variant, pair.TaskName)
+		mergeTaskDependencies = append(mergeTaskDependencies, task.Dependency{TaskId: depID, Status: evergreen.TaskSucceeded})
+	}
+
+	// update task in the database
+	mergeTaskID := execTaskIDTable.GetId(evergreen.MergeTaskVariant, evergreen.MergeTaskName)
+	return task.UpdateOne(bson.M{task.IdKey: mergeTaskID}, bson.M{"$set": bson.M{task.DependsOnKey: mergeTaskDependencies}})
```

Comment:
combine previous two statments, maybe on multiple lines and wrap? 

maybe it makes sense to write a function for this specific update operation as a function in the task package?

Reply:
done and done.